working-groups/query-standardization/dsl-survey/google-monitoring-query-language/README.md

```diff
@@ -0,0 +1,117 @@
+     understand what it means. This really increases the learning curve for both reading and writing. We 
+     encourage very little usage of MQL's syntactic sugar, and the sugar we do recommend can be applied uniformly to 
+     all queries in a standard way.
+
+     MQL's goal of achieving perfect suitability for Monarch was too narrow in scope. Despite the many 
+     good decisions made by early Monarch engineers, Monarch of course has some quirks (e.g. no null values, 
+     confusing semantics about stream kinds), and these are exposed directly to users of MQL. MQL could have acted 
+     as a shim layer that hides some of these quirks, but it doesn't. Ideally, a good observability language would 
+     be designed without a particular engine in mind.
+
+1. **Is the language intended at-rest data and/or streaming data?**:
+
+   At rest, but it could easily support streaming data.
+
+1. **Can users specify a source in the query language? E.g. a specific table or database?**:
+
+   Table, yes. Database, no, although this could be extended.
+
+   1. **Can users join with other sources than the primary database? E.g. a CSV file, cloud databases, etc.?**:
+
+      Not currently, but this could be supported.
+
+   1. **If joining of various time series sources is available, how are differing retentions handled?**:
+
+      This is not handled at the query level. Different processes manage removing data that is out of retention so 
+      that the query path can remain ignorant of this aspect.
+
+1. **How tightly coupled is the DSL to the data store model? (1 to 10 with 10 being extremely) tightly coupled)**:
+
+   2 or 3. There are some aspects of the language that are more coupled than I'd like (e.g. handling of stream kinds), 
+   but otherwise this language makes only a few assumptions about the data store model.
+
+   1. **Is the DSL flexible enough to operate on data in different storage formats or contexts?**:
+
+      Storage formats, yes.
+
+1. **What character sets are supported by the DSL?**:
+
+   ASCII for syntax, UTF-8 for data identifiers.
+
+   1. **What characters are special or reserved in the DSL**:
+
+      Common mathematical characters with support for quoting. See the 
+      [MQL Lexical Grammar](https://cloud.google.com/monitoring/mql/reference#lexical-elements).
+
+1. **Does the DSL allow for writing data or configuring the backing store or is it for querying only?**:
+
+   The language itself only supports querying, but Monarch is able to write query results back to the data store; this 
+   cannot be done on an ad-hoc basis and needs to be configured. MQL has no support for modifying configuration.
+
+## References
+
+* [MQL Overview](https://cloud.google.com/monitoring/mql)
+* [About the MQL Language](https://cloud.google.com/monitoring/mql/query-language)
+* [Monarc: Google’s Planet-Scale In-Memory Time Series Databaseh](https://www.vldb.org/pvldb/vol13/p3181-adams.pdf)
```

Comment:
Small typo:
Monarc**h**: Google’s Planet-Scale In-Memory Time Series Database~~h~~